server/issue_test.go

```diff
@@ -279,6 +280,255 @@ func TestRouteShareIssuePublicly(t *testing.T) {
+					Issue: jira.Issue{
+						Fields: &jira.IssueFields{
+							Creator: &jira.User{},
+						},
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			client, connection, error := tt.wh.fetchConnectedUser(
+				p,
+				tt.instanceID,
+			)
+			assert.Equal(t, tt.connection, connection)
+			assert.Equal(t, tt.client, client)
+			if tt.expectedErr != nil {
+				assert.Error(t, tt.expectedErr, error)
+			}
+		})
+	}
+}
+
+func TestApplyReporterNotification(t *testing.T) {
+	api := &plugintest.API{}
+
+	api.On("LogError", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
+
+	api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
+	p := Plugin{}
+	p.SetAPI(api)
+	p.instanceStore = p.getMockInstanceStoreKV(1)
+	p.userStore = getMockUserStoreKV()
+	wh := &webhook{
+		eventTypes: map[string]bool{createdCommentEvent: true},
+		JiraWebhook: &JiraWebhook{
+			Comment: jira.Comment{
+				UpdateAuthor: jira.User{},
+			},
+			Issue: jira.Issue{
+				Key: "TEST-10",
+				Fields: &jira.IssueFields{
+					Type: jira.IssueType{
+						Name: "Story",
+					},
+					Summary: "",
+				},
+				Self: "Abc",
+			},
+		},
+		notifications: []webhookUserNotification{},
+	}
+	tests := map[string]struct {
+		instanceID types.ID
+		reporter   *jira.User
+	}{
+		"Unable to load instance": {
+			instanceID: "instanceID",
+			reporter:   &jira.User{},
+		},
+		"No reporter": {
+			instanceID: testInstance1.InstanceID,
+			reporter:   nil,
+		},
+		"Success": {
+			instanceID: testInstance1.InstanceID,
+			reporter:   &jira.User{},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			totalNotifications := len(wh.notifications)
+			p.applyReporterNotification(
+				wh,
+				tt.instanceID,
+				tt.reporter,
+			)
+			if tt.reporter == nil || tt.instanceID == "instanceID" {
+				assert.Equal(t, len(wh.notifications)-totalNotifications, 0)
+			} else {
+				assert.Equal(t, len(wh.notifications)-totalNotifications, 1)
+			}
+		})
+	}
+}
+
+func TestGetUserSetting(t *testing.T) {
+	api := &plugintest.API{}
+
+	api.On("LogError", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
+
+	api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
+	p := Plugin{}
+	p.SetAPI(api)
+	p.instanceStore = p.getMockInstanceStoreKV(1)
+	p.userStore = getMockUserStoreKV()
+
+	tests := map[string]struct {
+		wh            *webhook
+		instanceID    types.ID
+		jiraAccountID string
+		jiraUsername  string
+		connection    *Connection
+		expectedErr   error
+	}{
+		"Unable to load instance": {
+			wh:            &webhook{},
+			instanceID:    "instanceID",
+			jiraAccountID: "",
+			jiraUsername:  "",
+			connection:    nil,
+			expectedErr:   errors.New("instance " + fmt.Sprintf("\"%s\"", "instanceID") + " not found"),
+		},
+		"Success": {
+			wh:            &webhook{},
+			instanceID:    testInstance1.InstanceID,
+			jiraAccountID: "",
+			jiraUsername:  "",
```

Comment:
both username and accountID is empty in all test cases make them constant and have them equal to mock value not empty